pkg/core/resources/apis/mesh/deprecated.go

```diff
@@ -5,15 +5,18 @@ import (
 	"strings"
 
 	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+
+	"github.com/kumahq/kuma/pkg/core/resources/model"
 )
 
 func (t *MeshResource) Deprecations() []string {
 	var deprecations []string
 
-	allErrs := apimachineryvalidation.NameIsDNS1035Label(t.GetMeta().GetName(), false)
+	name := model.GetDisplayName(t.GetMeta())
+	allErrs := apimachineryvalidation.NameIsDNS1035Label(name, false)
 	if len(allErrs) != 0 {
-		nameDeprecationMsg := fmt.Sprintf("Name that doesn't conform DNS (RFC 1035) format is deprecated: %s",
-			strings.Join(allErrs, "; "))
+		nameDeprecationMsg := fmt.Sprintf("Invalid name: '%s'. It does not conform to the DNS format (RFC 1035). This is deprecated. Errors: %s",
+			name, strings.Join(allErrs, "; "))
```

Comment:
What about adding the resource type?

```suggestion
		nameDeprecationMsg := fmt.Sprintf("Invalid %s resource name: '%s'. It does not conform to the DNS format (RFC 1035). This is deprecated. Errors: %s",
			MeshResourceTypeDescriptor.Name, name, strings.Join(allErrs, "; "))
```